library/core/src/ops/range.rs

```diff
@@ -147,6 +147,44 @@ impl<Idx: PartialOrd<Idx>> Range<Idx> {
     }
 }
 
+impl<Idx: core::cmp::Ord> Range<Idx> {
+    /// Compares the range to a scalar. Returns `Ordering::Equal`,
+    /// if the range contains the scalar. If not, returns `Ordering::Less`
+    /// or `Ordering::Greater`, depending on whether the range is
+    /// below or above the scalar.
+    ///
+    /// # Examples
```

Comment:
I would suggest also having some much simpler examples here too, like
```rust
assert_eq!((1..10).cmp_scalar(-1234), Ordering::Less);
assert_eq!((1..10).cmp_scalar(0), Ordering::Less);
assert_eq!((1..10).cmp_scalar(1), Ordering::Equal);
assert_eq!((1..10).cmp_scalar(5), Ordering::Equal);
assert_eq!((1..10).cmp_scalar(9), Ordering::Equal);
assert_eq!((1..10).cmp_scalar(10), Ordering::Greater);
assert_eq!((1..10).cmp_scalar(123456), Ordering::Greater);
```

Reply:
Or did I get those backwards?  All the more reason to have some examples like this ;)

Reply:
Indeed, I'll add them.

Reply:
Added!